### PR TITLE
Move worst subset selection from Ranker to the QuantizationAccuracyRestorer

### PR DESCRIPTION
**Pull request by andrey-churkin**
### Changes

Move worst subset selection from Ranker to the QuantizationAccuracyRestorer

### Reason for changes

This subset is required for tuning of hyperparameters

### Related tickets

N/A

### Tests

N/A

### CODE REVIEW

**Comment by andrey-churkin** [comment]
run pytorch pre-commit tests

File: nncf/quantization/algorithms/accuracy_control/algorithm.py [hunk 1]
```diff
@@ -94,6 +97,35 @@ def calculate_accuracy_drop(
     return should_terminate, accuracy_drop
 
 
+def select_worst_subset(
```
Comment (alexsu52) [inline nncf/quantization/algorithms/accuracy_control/algorithm.py:100]:
```suggestion
def  select_subset(
```

File: nncf/quantization/algorithms/accuracy_control/ranker.py [hunk 2]
```diff
@@ -80,33 +80,23 @@ class Ranker:
 
     def __init__(
         self,
-        ranking_subset_size: int,
```
Comment (alexsu52) [inline nncf/quantization/algorithms/accuracy_control/ranker.py:83]: It looks like, this parameter was relevant for the `Ranker`.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

**Comment by andrey-churkin** [comment]
run pylint pre-commit tests
run pytorch pre-commit tests
run tensorflow pre-commit tests
run openvino pre-commit tests
run onnx pre-commit tests

File: nncf/quantization/algorithms/accuracy_control/ranker.py (hunk 2 above)
Comment (andrey-churkin) [reply nncf/quantization/algorithms/accuracy_control/ranker.py]: Fixed

File: nncf/quantization/algorithms/accuracy_control/algorithm.py (hunk 1 above)
Comment (andrey-churkin) [reply nncf/quantization/algorithms/accuracy_control/algorithm.py]: Done

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**